backend/src/shared/shared.module.ts

```diff
@@ -12,7 +14,33 @@ import { JwtModule } from '@nestjs/jwt';
 			}),
 			inject: [ConfigService],
 		}),
+		MulterModule.registerAsync({
```

Comment:
I added a comment above the code with some explanation, and I refactored the code to be more readable (because it was bad..)

currently the `filename` method changes the filename to `u{user_id}_avatar.{extension}` 
just so they all have a clear format in filenames

the `destination` method sets the dest to either ./public or ./public/avatar (can be more added if we need more upload directories)

There are more options to multer but they are unused for now

Reply:
Awesome thank youuuu!